Update .gitignore

Blunderchips opened the pull request:
Excluded Intellij project files. For more things that could added check out the libGDX generated gitignore.
The rational behind this change is that I would like to take another look at MochaDoom and Intellij is my current IDE of choice.

Blunderchips committed:
Update .gitignore

Excluded Intellij project files

AXDOOMER commented on .gitignore:
```diff
@@ -9,3 +9,8 @@
+# Intellij 
+*.iml
+.idea/
+/out/
```
Please change `/out/` for `out/`.  The excluded directories already specified in this file contain a mistake, because directories shouldn't start with a `/`. The ending `/` is required though. 

Please also remove the front `/` for the other directories in the same time. 

```
/nbproject/private/
/build/
/dist/
/nbproject/
```

Thanks!

Blunderchips replied:
Will do

AXDOOMER commented:
libGDX's [.gitignore](https://github.com/libgdx/libgdx/blob/master/.gitignore) has directories that start with a `/`, but this isn't how it should be done. It may still work though. 

For a better examples, check out the site https://www.gitignore.io/.  Search for `Intellij` or `Visual Studio` for examples about IDEs.

Blunderchips committed:
Update .gitignore

Made requested change to directory paths.
Added gitignore GitHub recommends for java projects.

AXDOOMER commented:
Also, please use descriptive names for your commits. For example, if you'd have named this one 
"_Update .gitignore: Excluded Intellij project files_", we would have a better idea about what has changed from the commit name. 

I don't have a contribution guide yet, but if you want to contribute back, I can merge your changes without asking for changes if your they are top-notch. 

Have a nice week-end :-)

Blunderchips commented:
I have made the requested change and added some more general exclusions as suggested by GitHub.

AXDOOMER commented:
Alright, thanks!!

Blunderchips commented:
> 
> 
> Also, please use descriptive names for your commits. For example, if you'd have named this one
> "_Update .gitignore: Excluded Intellij project files_", we would have a better idea about what has changed from the commit name.
> 
> I don't have a contribution guide yet, but if you want to contribute back, I can merge your changes without asking for changes if your they are top-notch.
> 
> Have a nice week-end :-)

Thanks, will do in the future.
